libnavui-maps/src/main/java/com/mapbox/navigation/ui/map/route/api/MapboxMapRouteLineApi.kt

```diff
@@ -0,0 +1,137 @@
+package com.mapbox.navigation.ui.map.route.api
+
+import android.content.Context
+import androidx.annotation.WorkerThread
+import androidx.core.content.ContextCompat
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.LineString
+import com.mapbox.navigation.ui.base.map.route.api.MapRouteLineApi
+import com.mapbox.navigation.ui.base.map.route.model.RouteLineOptions
+import com.mapbox.navigation.ui.base.map.route.model.RouteLineState
+import com.mapbox.navigation.ui.maps.R
+
+object MapboxMapRouteLineApi : MapRouteLineApi {
+
+    override fun createDefaultOptions(context: Context) = RouteLineOptions.Builder()
```

Comment:
What if instead of passing a context we abstract it by passing a color provider function. Something like colorProvider: (resource: Int) -> Int?

Reply:
Also, based on our discussion yesterday, did we agree on converting this `object` to `interface` so that we can later compose them for being used by the engine?
If yes, then can we change this to an `interface`?
cc @LukasPaczos @cafesilencio

Reply:
@abhishek1508 this PR already contains the `MapRouteLineApi` interface and its `MapboxMapRouteLineApi` `object`-based implementation.